backend/controllers/sprints.js

```diff
@@ -2,37 +2,32 @@ const express = require('express')
 const { checkLogin } = require('../middleware')
 
 const sprintsRouter = express.Router()
+const db = require('../models/index')
 
-let mockSprints = [
-  {
-    id: 1,
-    start_date: '2024-02-01',
-    end_date: '2024-02-14',
-    sprint: 1,
-    user_id: '8668e87f3f7cbff3067731ed4a181879949716608e4fa93a9ded969d1d2626f3',
-  },
-  {
-    id: 2,
-    start_date: '2024-02-15',
-    end_date: '2024-02-28',
-    sprint: 2,
-    user_id: '8668e87f3f7cbff3067731ed4a181879949716608e4fa93a9ded969d1d2626f3',
-  },
-  {
-    id: 3,
-    start_date: '2024-02-29',
-    end_date: '2024-03-06',
-    sprint: 3,
-    user_id: '8668e87f3f7cbff3067731ed4a181879949716608e4fa93a9ded969d1d2626f3',
-  },
-  {
-    id: 4,
-    start_date: '2024-02-09',
-    end_date: '2024-02-16',
-    sprint: 1,
-    user_id: '10f8bdef82c62acf57da2c7bf8064641fb14f7b07f49b3f2f3316bf697ea3706',
+
+// latest group, because student may belong to multiple groups (dropout, etc.)
+const findLatestGroupId = async (studentNumber) => {
+  try {
+    const latestGroup = await db.Group.findOne({
+      include: [{
+        model: db.User,
+        as: 'students',
+        where: { student_number: studentNumber }
+      }],
+      order: [['createdAt', 'DESC']]
+    })
+    if (!latestGroup) {
```

Comment:
Is this if statement needed? Sequelize findOne method (may?) return null if no matches. In that case line 23 should throw TypeError. 

This is not a big deal but same lines are repeated at the moment (20, 21 -> 26, 27)

Reply:
Yes and no depending on point of view: it is correct that line 23 would throw TypeError if latestGroup is null. But imo, it is better practice to raise controlled errors and custom error message when you know what is wrong. This makes debugging easier, and in a sense separates uncontrolled and unexpected bugs from controlled error barriers. Prefer to leave as it is.

Yes, the error message should not be the same, these are different cases. The first inside try is that user doesn't belong to a group. One in catch part is for whatever else could have gone wrong. Changed the latter message.